File: UniBlock/src/main/scala/helper/StringHelper.scala
package it.unifi.nave.uniblock
package helper

import helper.StringHelper.PADDING

object StringHelper {
  val LINE_LENGTH = 100
  val PADDING = "-"
  val MARGIN = 5
  val FIELD_LENGTH = 20

  def emptyLine: String = PADDING.repeat(LINE_LENGTH)

  def formatCenter(string: String): String = {
    val normalized = if (string.length % 2 == 0) string else string + PADDING
    val toFill = LINE_LENGTH - string.length
    PADDING.repeat(toFill / 2) + normalized + PADDING.repeat(toFill / 2)
  }

  def formatTitle(title: String): String = {
    s"""$emptyLine
       |${formatCenter(s" ${title.toUpperCase} ")}
       |$emptyLine""".stripMargin
  }

  def formatLeft(any: Any, label: String): String = formatLeft(any.toString, label)

  def formatLeft(string: String, label: String): String = {
    val margin = PADDING.repeat(MARGIN)
    val fieldName = s"$margin${(" " + label + " ").padTo(FIELD_LENGTH, PADDING(0))} = "
    val leftMargin = PADDING.repeat(fieldName.length - 1) + " "
    val realLine = LINE_LENGTH - fieldName.length - MARGIN - 1
    val rightMargin = " " + (if (string.length % realLine != 0) PADDING.repeat(realLine - string.length % realLine) else "") + margin
    string.grouped(realLine).mkString(fieldName, s" $margin\n$leftMargin", rightMargin)
  }
}

File: src/main/java/it/unifi/nave/uniblock/helper/StringHelper.java
package it.unifi.nave.uniblock.helper;

// TODO Rivedere con commons
public class StringHelper {
    private static final int LINE_LENGTH = 100;
    private static final String PADDING = "-";
    private static final int MARGIN = 5;
    private static final int FIELD_LENGTH = 20;

    public static String emptyLine() {
        return PADDING.repeat(LINE_LENGTH);
    }

    public static String formatCenter(String string) {
        var normalized = (string.length() % 2 == 0) ? string : string + PADDING;
        var toFill = LINE_LENGTH - string.length();
        return PADDING.repeat(toFill / 2) + normalized + PADDING.repeat(toFill / 2);
    }

    public static String formatTitle(String title) {
        return emptyLine() + "\n"
                + formatCenter(title.toUpperCase()) + "\n"
                + emptyLine();
    }

    public static String formatLeft(Object object, String label) {
        return formatLeft(object.toString(), label);
    }

    public static String formatLeft(String string, String label) {
        return null;
//        var margin = PADDING.repeat(MARGIN);
//        var fieldName = margin + Strings.padEnd(" " + label + " ", FIELD_LENGTH, PADDING.charAt(0)) + " = ";
//        var leftMargin = PADDING.repeat(fieldName.length() - 1) + " ";
//        var realLine = LINE_LENGTH - fieldName.length() - MARGIN - 1;
//        var rightMargin = " " + ((string.length() % realLine != 0) ? PADDING.repeat(realLine - string.length() % realLine) : "") + margin;
//        return Splitter.fixedLength(realLine).splitToStream(string).collect(Collectors.joining(" " + margin + "\n" + leftMargin, fieldName, rightMargin));
    }
}
